port_ocean/utils/signal.py

```diff
@@ -0,0 +1,57 @@
+import signal
+from typing import Callable
+
+from loguru import logger
+from werkzeug.local import LocalProxy, LocalStack
+
+from port_ocean.utils.misc import generate_uuid
+
+
+class SignalHandler:
+    def __init__(self):
+        self._handlers: dict[str, Callable] = {}
```

Comment:
Not especially related to the signal class, but I don't see that there is signal handling for all the event listeners ( kafka, once )